src/QsCompiler/Tests.Compiler/TestCases/OptimizerTests/Arithmetic_input.qs

```diff
@@ -11,4 +11,9 @@ namespace Microsoft.Quantum.Testing.Optimization.Arithmetic {
         let a = w ? 2 | y / 2;
         return a;
     }
+
+    operation Test2 (x : Int) : Int {
```

Comment:
It would be a really good idea to have tests for other data types as well - I know we are a bit thin on tests in this part of the code in general, but that's one of the reasons why these pieces of code are considered highly experimental.

Reply:
Added more test cases to cover `Double` and `BigInt`.